packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -77,13 +89,53 @@ abstract contract FundingRateApplier is Testable, Lockable {
      * @notice Constructs the FundingRateApplier contract. Called by child contracts.
      * @param _finderAddress Finder used to discover financial-product-related contracts.
      */
-    constructor(address _finderAddress) public {
-        finder = FinderInterface(_finderAddress);
+    constructor(
+        FixedPoint.Unsigned memory _fundingRateBondPercentage,
+        FixedPoint.Unsigned memory _fundingRateRewardRate,
+        bytes32 _fundingRateIdentifier,
+        address _collateralAddress,
+        address _finderAddress,
+        address _timerAddress
+    ) public FeePayer(_collateralAddress, _finderAddress, _timerAddress) {
+        fundingRate.updateTime = getCurrentTime();
+
+        // Seed the cumulative multiplier as 1, from which it will be scaled as funding rates are applied over time.
+        fundingRate.cumulativeMultiplier = FixedPoint.fromUnscaledUint(1);
         emergencyShutdownTimestamp = 0;
-        lastUpdateTime = getCurrentTime();
 
-        // Seed the initial funding rate in the cumulativeFundingRateMultiplier 1.
-        cumulativeFundingRateMultiplier = FixedPoint.fromUnscaledUint(1);
+        fundingRate.rewardRate = _fundingRateRewardRate;
+        fundingRate.identifier = _fundingRateIdentifier;
+        fundingRate.bondPercentage = _fundingRateBondPercentage;
+    }
+
+    function proposeNewRate(FixedPoint.Signed memory rate, uint256 timestamp)
+        external
+        nonReentrant()
+        updateFundingRate()
+    {
+        require(fundingRate.proposalTime == 0, "Proposal in progress");
+
+        // Timestamp must be after the last funding rate update time, within the last 30 minutes, and it cannot be more
+        // than 90 seconds ahead of the block timestamp.
```

Comment:
This comment isn't consistent with the inequality, either the inequality should be `timestamp <= currentTime.add(90)` or the comment should read "it cannot be 90 seconds or more ahead of the block timestamp"

Reply:
Ah, I see. Fixed.